### PR TITLE
feat(pcc.user): add new active directory

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/pcc-federation` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-10027 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
I added the section add new active directory.

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.controller.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import {
+  FIELD_NAME,
+  GUIDE_FEDERATION,
+  PLACEHOLDER,
+} from './federation-add.constant';
+
+export default class {
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.controller.js:7]: **nitpick:**
```suggestion
export default class FederationAddCtrl{
```

**Review by anooparveti — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.controller.js (hunk 1 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.controller.js]: It's applied

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.constant.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  AU: 'https://docs.ovh.com/au/en/private-cloud/federation/',
+  SG: 'https://docs.ovh.com/sg/en/private-cloud/federation/',
+  FR: 'https://docs.ovh.com/fr/private-cloud/federation/',
+  WE: 'https://docs.ovh.com/us/en/private-cloud/federation/',
+  WS: 'https://docs.ovh.com/us/es/private-cloud/federation/',
+  US: 'https://docs.ovh.com/us/en/private-cloud/federation/',
+};
+
+export const PLACEHOLDER = {
+  username: 'vcenterbind@example.com',
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.constant.js:22]: userName

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.constant.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  ldapHostname: 'ad.example.com',
+  password: '************',
+  ldapTcpPort: '636',
+  ip: '203.0.113.78',
+};
+
+const FIELD = 'Field';
+
+export const FIELD_NAME = {
+  username: `userName${FIELD}`,
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.constant.js:38]: Can you use pascal name here to be constant with others ? userName

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.constant.js (hunk 2 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.constant.js]: It's applied. It was a confusion with api parameter that isn't in pascal name.

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.constant.js (hunk 3 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.constant.js]: It's applied. It was a confusion with api parameter that isn't in pascal name.

**Review by sachinrameshn — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.controller.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    this.model = {
+      domainName: null,
+      domainAlias: null,
+      description: null,
+      ip: null,
+      ldapHostname: null,
+      ldapTcpPort: null,
+      sslThumbprint: null,
+      username: null,
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.controller.js:30]: userName ?

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.html [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+            disabled="$ctrl.loaders.submitting || $ctrl.loaders.submitted || form.$invalid"
+        ></oui-form-actions>
+    </div>
+</form>
+
+<div
+    data-ng-if="$ctrl.loaders.submitted"
+    class="modal"
+    role="dialog"
+    style="display: block;"
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.html:205]:
```suggestion
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.html (hunk 5 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.html]: It's applied I changed with d-block in class balise

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.controller.js (hunk 4 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/add/federation-add.controller.js:30]: In the api it is username without capital letter.

**Review by anooparveti — APPROVED** [review]

**Pull request merged**